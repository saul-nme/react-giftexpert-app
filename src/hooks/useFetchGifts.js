import { useEffect, useState } from "react";
import { getGift } from "../helpers/getGifts";


export const useFetchGifts = (category) => {

   const [state, setState] = useState({
      data: [],
      loading: true,
   });

   useEffect(() => {
      getGift(category) .then((images) => {
         setState(() => ({
            data: images,
            loading: false,
         }));
      });
   }, [ category ]);

   return state;

}
