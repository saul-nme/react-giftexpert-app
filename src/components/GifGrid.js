import React from 'react'
import { useFetchGifts } from '../hooks/useFetchGifts'
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types'

export default function GifGrid({ category }) {

   const { data: images, loading } = useFetchGifts(category);

   return (
      <>
         <h3>{category}</h3>
         {loading && "Loading..."}
         <div className="card-grid ">
            {
               images.map((img) => (
                  <GifGridItem
                     key={img.id}
                     {...img}
                  />
               ))
            }
         </div>
      </>
   )
}

GifGrid.propTypes = {
   category: PropTypes.string.isRequired
}