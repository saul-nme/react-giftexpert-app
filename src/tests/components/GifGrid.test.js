import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifts } from "../../hooks/useFetchGifts";

jest.mock('../../hooks/useFetchGifts')

describe('Pruebas en <GifGrid/>', () => {
   const category = "coso";

   test('Debe mostrarse correctamente', () => {
      useFetchGifts.mockReturnValue({
         data: [],
         loading: true,
      });

      const wrapper = shallow(<GifGrid category={category} />);
      expect(wrapper).toMatchSnapshot();
   });

   test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifts', () => {
      const gifs = [{
         id: "ABC",
         url: 'https://localhost/coso.png',
         title: 'Cosos'
      }]

      useFetchGifts.mockReturnValue({
         data: gifs,
         loading: false,
      });

      const wrapper = shallow(<GifGrid category={category} />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('p').exists()).toBe(false);
      expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

   });

});
