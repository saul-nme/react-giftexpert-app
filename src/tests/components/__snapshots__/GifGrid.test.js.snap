// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    coso
  </h3>
  Loading...
  <div
    className="card-grid "
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid/> debe de mostrar items cuando se cargan im√°genes useFetchGifts 1`] = `
<Fragment>
  <h3>
    coso
  </h3>
  <div
    className="card-grid "
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cosos"
      url="https://localhost/coso.png"
    />
  </div>
</Fragment>
`;
